src/core/toga/sources/row.py

```diff
@@ -0,0 +1,50 @@
+
+class Row:
+    def __init__(self, **data):
+        self._attrs = list(data.keys())
+        self._source = None
+        for name, value in data.items():
+            setattr(self, name, value)
+
+    ######################################################################
+    # Utility wrappers
+    ######################################################################
+
+    def __setattr__(self, attr, value):
+        super().__setattr__(attr, value)
+        if attr in self._attrs and self._source is not None:
+            self._source._notify('change', item=self)
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.as_dict() == other.as_dict()
+
+    def __repr__(self):
+        return repr(self.as_dict())
```

Comment:
This should include an identifier so we can easily tell the difference between a Row object and a dict that contains the same items.